packages/daimo-api/src/router.ts

```diff
@@ -283,14 +289,25 @@ export function createRouter(
               args: [ephemeralNotesAddress, maxUint256],
             }),
           },
+          {
+            // Approve paymaster contract infinite spending on behalf of the account
+            dest: tokenMetadata.address,
+            value: 0n,
+            data: encodeFunctionData({
+              abi: erc20ABI,
+              functionName: "approve",
+              args: [tokenMetadata.paymasterAddress, maxUint256],
+            }),
+          },
```

Comment:
does everyone need to reroll their accounts (in which case we can remove the v6 > v7 migration) or is there a mechanism to just-in-time approve?

thoughts
- reroll seems fine
- eventually / soon we may need an approval mechanism that works like this: list of approved contracts somewhere. all auto-approved on account deploy. then in sync, older accounts learn about the ones they still have yet to approve and do it as part of the next transaction.

Reply:
i think we should reroll soon anyway since so many changes are coming: nameregistry upgradeability changes for example are probably not backwards compatible anyway.

Reply:
dropped v5 and v6